services/subscriptions/src/main/java/org/activiti/cloud/services/notifications/graphql/subscriptions/datafetcher/AbstractDestinationResolver.java

```diff
@@ -71,9 +78,107 @@ protected String resolveFieldName(DataFetchingEnvironment environment) {
 	}
 	
 
-	private Optional<String> resolveArgument(DataFetchingEnvironment environment, String arumentName) {
-	    return Optional.ofNullable(environment.getArgument(arumentName));
+	private List<String> resolveArgument(DataFetchingEnvironment environment, String arumentName) {
+	    List<String> value = new ArrayList<>();
+	    
+	    Object argument = environment.getArgument(arumentName);
+	    
+	    if(argument instanceof List)
+	        value.addAll(environment.getArgument(arumentName));
+	    else if(argument != null )
+	        value.add(argument.toString());
+	    else
+	        value.add(wildcard());
+	    
+	    
+	    return value;
 	}
 
 
+	public static <T> List<List<T>> crossJoin(List<List<T>> factors) {
+	    return new CartesianProduct<T>(factors).stream()
+	                                           .collect(Collectors.toList());
+	}
+	
+	public static List<List<String>> zip(List<String> list1, List<String> list2 ) {
+	    return crossJoin(Arrays.asList(list1, list2));
+	}
+
+    static class CartesianProduct<T> implements Iterable<List<T>> {
+        private final Iterable<? extends Iterable<T>> factors;
+
+        public CartesianProduct(final Iterable<? extends Iterable<T>> factors) {
+            this.factors = factors;
+        }
+
+        @Override
+        public Iterator<List<T>> iterator() {
+            return new CartesianProductIterator<>(factors);
+        }
+        
+        public Stream<List<T>> stream() {
+            return StreamSupport.stream(new CartesianProduct<>(factors).spliterator(), false);
+        }
+    }
+
+    static class CartesianProductIterator<T> implements Iterator<List<T>> {
+        private final List<Iterable<T>> factors;
+        private final Stack<Iterator<T>> iterators;
+        private final Stack<T> current;
+        private List<T> next;
+        private int index = 0;
+
+        private void computeNext() {
+            while (true) {
+                if (iterators.get(index).hasNext()) {
+                    current.add(iterators.get(index).next());
+                    if (index == factors.size() - 1) {
+                        next = new ArrayList<>(current);
+                        current.pop();
+                        return;
+                    }
+                    index++;
+                    iterators.add(factors.get(index).iterator());
+                } else {
+                    index--;
+                    if (index < 0) {
+                        return;
+                    }
+                    iterators.pop();
+                    current.pop();
+                }
+            }
+        }
+
+        public CartesianProductIterator(final Iterable<? extends Iterable<T>> factors) {
```

Comment:
Could you move constructor up, before any other method?

Reply:
fixed